pytorch_lightning/core/saving.py

```diff
@@ -125,10 +126,11 @@ def load_from_checkpoint(
             pretrained_model.freeze()
             y_hat = pretrained_model(x)
         """
-        if map_location is not None:
-            checkpoint = pl_load(checkpoint_path, map_location=map_location)
-        else:
-            checkpoint = pl_load(checkpoint_path, map_location=lambda storage, loc: storage)
+        with pl_legacy_patch():
+            if map_location is not None:
+                checkpoint = pl_load(checkpoint_path, map_location=map_location)
```

Comment:
@awaelchli quick question: would see any value to bring this context manager directly inside pl_load ?

Reply:
yes, there may be a minor value, however I think we might want to keep the ones in the checkpoint connector anyway since there we load using the training type plugin / checkpointIO plugin and if a custom checkpoint io plugin does not call pl_load it would need to remember calling `pl_legacy_patch`